tests/wpunit/AccessFunctionsTest.php

```diff
@@ -701,4 +701,380 @@ public function testRegisterMutationWithIsPrivatePreventsPublicMutationExecution
+		$this->assertCount( 1, $actual['data']['contentNodes']['nodes'] );
+	}
+
+	public function testDeregisterInputType() {
+		deregister_graphql_type( 'DateQueryInput' );
+
+		// Ensure the schema is still queryable.
+		$query = '
+		{
+			__schema {
+				types {
+					name
+				}
+			}
+		}
+		';
+
+		$actual = graphql( compact( 'query' ) );
+
+		$this->assertIsValidQueryResponse( $actual );
+		$this->assertArrayNotHasKey( 'errors', $actual );
+		$this->assertNotContains( 'DateQueryInput', array_column( $actual['data']['__schema']['types'], 'name' ) );
+
+		// Ensure the input is removed from the schema.
+		$query = '
+		{
+			__type(name: "RootQueryToPostConnectionWhereArgs") {
+				inputFields {
+					name
+				}
+			}
+		}';
+
+		$actual = $this->graphql( compact( 'query' ) );
+
+		$this->assertIsValidQueryResponse( $actual );
+		$this->assertArrayNotHasKey( 'errors', $actual );
+		$this->assertNotContains( 'dateQuery', array_column( $actual['data']['__type']['inputFields'], 'name' ) );
+
+		// Ensure query still works
+		$post_id = $this->factory()->post->create(
+			[
+				'post_title' => 'Test deregister input type',
+				'post_status' => 'publish',
+			]
+		);
+
+		$query = '
+		{
+			posts {
+				nodes {
+					id
+				}
+			}
+		}
+		';
+
+		$actual = $this->graphql( compact( 'query' ) );
+
+		$this->assertIsValidQueryResponse( $actual );
+		$this->assertArrayNotHasKey( 'errors', $actual );
+		$this->assertCount( 1, $actual['data']['posts']['nodes'] );
+	}
+
+	public function testDeregisterInterfaceType() {
+		deregister_graphql_type( 'NodeWithTitle' );
+
+		// Ensure the schema is still queryable.
+		$query = '
+		{
+			__schema {
+				types {
+					name
+				}
+			}
+		}
+		';
+
+		$actual = graphql( compact( 'query' ) );
+
+		$this->assertIsValidQueryResponse( $actual );
+		$this->assertArrayNotHasKey( 'errors', $actual );
+		$this->assertNotContains( 'NodeWithTitle', array_column( $actual['data']['__schema']['types'], 'name' ) );
+
+		// Ensure the interface is removed from the schema.
+		$query = '
+		{
+			__type(name: "Post") {
+				interfaces {
+					name
+				}
+			}
+		}';
+
+		$actual = $this->graphql( compact( 'query' ) );
+
+		$this->assertIsValidQueryResponse( $actual );
+		$this->assertArrayNotHasKey( 'errors', $actual );
+		$this->assertNotContains( 'NodeWithTitle', array_column( $actual['data']['__type']['interfaces'], 'name' ) );
+
+		// Ensure query still works
+		$post_id = $this->factory()->post->create(
+			[
+				'post_title' => 'Test deregister interface type',
+				'post_status' => 'publish',
+			]
+		);
+
+		$query = '
+		{
+			posts {
+				nodes {
+					id
+				}
+			}
+		}';
+
+		$actual = $this->graphql( compact( 'query' ) );
+
+		$this->assertIsValidQueryResponse( $actual );
+		$this->assertArrayNotHasKey( 'errors', $actual );
+		$this->assertCount( 1, $actual['data']['posts']['nodes'] );
+	}
+
+	public function testDeregisterUnionType() {
+		deregister_graphql_type( 'MenuItemObjectUnion' );
```

Comment:
It would be better to test a non-deprecated Union type, but im not exactly sure what can be used.